src/test/java/TestShooterSubsystem.java

```diff
@@ -1,43 +1,18 @@
 import frc.robot.subsystems.shooterSubsystem.ShooterSubsystem;
 import frc.robot.subsystems.shooterSubsystem.ShooterSubsystem.SpeakerConfig;
-import java.util.Arrays;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
 
 public class TestShooterSubsystem {
-  // All distances in this test, aside from the negative distance and the extremely large distance,
-  // should be part of speakerConfigList! Changes to one should change the other.
-  // TODO: Make more sensible
   @Test
   public void testSpeakerConfig() {
-    SpeakerConfig speedsPair;
-    double[] speedsArray = {0, 0};
-    double[][] speakerConfigListTest = {
-      {-100, 2350, 1950},
-      {0, 2350, 1950},
-      {40, 350, 350},
-      {195, 500, 500},
-      {290, 700, 700},
-      {10000, 700, 700}
-    };
-    ;
-    double[] configSpeeds;
+    SpeakerConfig testConfig;
 
-    for (int i = 0; i < speakerConfigListTest.length; i++) {
-      speedsPair = ShooterSubsystem.calculateShooterSpeeds(speakerConfigListTest[i][0]);
-      speedsArray[0] = speedsPair.getSpeedLeftInRPM();
-      speedsArray[1] = speedsPair.getSpeedRightInRPM();
-      configSpeeds = arraySlice(speakerConfigListTest[i], 1, 2);
-      Assertions.assertArrayEquals(configSpeeds, speedsArray);
+    for (int i = 0; i < ShooterSubsystem.speakerConfigList.length; i++) {
+      testConfig =
+          ShooterSubsystem.calculateShooterSpeeds(
+              ShooterSubsystem.speakerConfigList[i].getDistanceInInches());
+      Assertions.assertEquals(testConfig, ShooterSubsystem.speakerConfigList[i]);
```

Comment:
This test only tests exact matches for Shooter Configurations.

We should also consider test cases:

- Is config list in sorted order, with both distance, left speed and right speed increasing as we move through the list?
- Does a distance value below minimum provide the lowest possible speeds?
- Does a distance value above maximum provide the highest possible speed?

Reply:
Tests added in 3cc5cbb